engines/bastion_katello/test/debs/debs.controller.test.js

```diff
@@ -0,0 +1,110 @@
+            this.setParams = function (params) {};
+            this.getParams = function (params) { return {}; };
+            this.refresh = function () {};
+            this.getAllSelectedResults = function () {};
+        };
+        Errata = {};
+
+/*
+ * TODO:
+        Task = {
+            registerSearch: function() {},
+            unregisterSearch: function () {}
+        };
+*/
+    });
+
+    beforeEach(inject(function(_$controller_, $rootScope, _$location_, MockResource, translateMock) {
+        Deb = MockResource.$new();
+        //Repository = MockResource.$new();
+        $scope = $rootScope.$new();
+        $location = _$location_;
+
+        $controller = _$controller_;
+        dependencies = {
+            $scope: $scope,
+            $location: $location,
+            Nutupane: Nutupane,
+            Deb: Deb,
+            //Task: Task,
+            //Repository: Repository,
+            CurrentOrganization: 'CurrentOrganization',
+            translate: translateMock
+        };
+
+        $controller('DebsController', dependencies);
+    }));
+
+    it('attaches the nutupane table to the scope', function() {
+        expect($scope.table).toBeDefined();
+    });
+
+
+/*
+ * TODO:
+    it('gets a list of deb repositories for the organization', function () {
+        expect($scope.repositories[0]).toBe($scope.repository);
+        expect($scope.repositories.length).toBe(2);
+    });
+
+    it('should have a list of repositories that include an all option', function () {
+        expect($scope.repositories[0]['id']).toBe('all');
+    });
+*/
+
+    it("allows the filtering of packages", function () {
+        $scope.showApplicable = false;
+        $scope.showUpgradable = false;
+        $scope.toggleFilters();
+        expect($scope.table.params['packages_restrict_applicable']).toBe(false)
+        expect($scope.table.params['packages_restrict_upgradable']).toBe(false)
+    });
+
+    it("ensures showApplicable is true if showUpgradable is true", function () {
+        $scope.showApplicable = false;
+        $scope.showUpgradable = true;
+        $scope.toggleFilters();
+        expect($scope.table.params['packages_restrict_applicable']).toBe(true)
+        expect($scope.table.params['packages_restrict_upgradable']).toBe(true)
+    });
+
+/*
+ * TODO:
+    it('should set the repository_id param on Nutupane when a repository is chosen', function () {
+        spyOn($scope.nutupane, 'setParams');
+        spyOn($scope.nutupane, 'refresh');
+
+        $scope.repository = {id: 1};
+        $scope.$apply();
+
+        expect($scope.nutupane.setParams).toHaveBeenCalledWith({'repository_id': 1});
+        expect($scope.nutupane.refresh).toHaveBeenCalled();
+    });
+
+    it('allows the setting of the repositoryId via a query string parameter', function () {
+        $location.search('repositoryId', '1');
+
+        $controller('DebsController', dependencies);
+
+        expect($scope.repository.id).toBe(1);
+    });
+*/
```

Comment:
I think all these commented out tests would be good to get working

Reply:
Seems to me like the Repository is not mocked.
Do I have to define something for this to work?
```
HeadlessChrome 90.0.4430 (Linux 0.0.0) Controller: DebsController gets a list of deb repositories for the organization FAILED
	TypeError: Cannot read property '0' of undefined
	    at UserContext.<anonymous> (/home/jenkins/workspace/katello-pr-test/engines/bastion_katello/test/debs/debs.controller.test.js:60:35)
HeadlessChrome 90.0.4430 (Linux 0.0.0) Controller: DebsController should have a list of repositories that include an all option FAILED
	TypeError: Cannot read property '0' of undefined
	    at UserContext.<anonymous> (/home/jenkins/workspace/katello-pr-test/engines/bastion_katello/test/debs/debs.controller.test.js:65:35)
HeadlessChrome 90.0.4430 (Linux 0.0.0) Controller: DebsController should set the repository_id param on Nutupane when a repository is chosen FAILED
	Error: <spyOn> : could not find an object to spy upon for setParams()
	Usage: spyOn(<object>, <methodName>)
	    at UserContext.<anonymous> (/home/jenkins/workspace/katello-pr-test/engines/bastion_katello/test/debs/debs.controller.test.js:85:9)
HeadlessChrome 90.0.4430 (Linux 0.0.0) Controller: DebsController allows the setting of the repositoryId via a query string parameter FAILED
	TypeError: Cannot read property 'id' of undefined
	    at UserContext.<anonymous> (/home/jenkins/workspace/katello-pr-test/engines/bastion_katello/test/debs/debs.controller.test.js:100:34)

Reply:
see https://github.com/Katello/katello/pull/8277#discussion_r713111454